### PR TITLE
Fix partition's gc

### LINKED ISSUE #27680: [Bug]: Calling a step_task infinite times: report there is no value on key =... collection not found.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-cb71a3e2359215ada0372439d3f91547f88873b3
- Deployment mode(standalone or cluster): standalone
- MQ type(rocksmq, pulsar or kafka): pulsar
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus
- OS(Ubuntu or CentOS): MacOS
- CPU/Memory: M1 16G
- GPU: 
- Others:
```

### Current Behavior

Execute infinite times to release a not-existing collection.

### Expected Behavior

Do only once.

### Steps To Reproduce

```markdown
1. Start Milvus.
2. Run L0 L1 e2e tests.
3. Stop Milvus.
```

### Milvus Log

```
[2023/10/12 19:33:24.795 +08:00] [WARN] [rootcoord/step_executor.go:75] ["failed to execute step, wait for reschedule"] [error="there is no value on key = by-dev/meta/root-coord/collection/444883324913245356: collection=444883324913245356: collection not found"] [errorVerbose="there is no value on key = by-dev/meta/root-coord/collection/444883324913245356: collection=444883324913245356: collection not found
(1) attached stack trace
  -- stack trace:
  | github.com/milvus-io/milvus/pkg/util/merr.WrapErrCollectionNotFound
  | \t/Users/zilliz/repo/github/chyezh/milvus/pkg/util/merr/utils.go:327
  | [...repeated from below...]
Wraps: (2) there is no value on key = by-dev/meta/root-coord/collection/444883324913245356
Wraps: (3) attached stack trace
  -- stack trace:
  | github.com/milvus-io/milvus/pkg/util/merr.wrapWithField
  | \t/Users/zilliz/repo/github/chyezh/milvus/pkg/util/merr/utils.go:681
  | github.com/milvus-io/milvus/pkg/util/merr.WrapErrCollectionNotFound
  | \t/Users/zilliz/repo/github/chyezh/milvus/pkg/util/merr/utils.go:325
  | github.com/milvus-io/milvus/internal/metastore/kv/rootcoord.(*Catalog).loadCollectionFromDb
  | \t/Users/zilliz/repo/github/chyezh/milvus/internal/metastore/kv/rootcoord/kv_catalog.go:194
  | github.com/milvus-io/milvus/internal/metastore/kv/rootcoord.(*Catalog).loadCollectionFromDefaultDb
  | \t/Users/zilliz/repo/github/chyezh/milvus/internal/metastore/kv/rootcoord/kv_catalog.go:207
  | github.com/milvus-io/milvus/internal/metastore/kv/rootcoord.(*Catalog).loadCollection
  | \t/Users/zilliz/repo/github/chyezh/milvus/internal/metastore/kv/rootcoord/kv_catalog.go:212
  | github.com/milvus-io/milvus/internal/metastore/kv/rootcoord.(*Catalog).DropPartition
  | \t/Users/zilliz/repo/github/chyezh/milvus/internal/metastore/kv/rootcoord/kv_catalog.go:524
  | github.com/milvus-io/milvus/internal/rootcoord.(*MetaTable).RemovePartition
  | \t/Users/zilliz/repo/github/chyezh/milvus/internal/rootcoord/meta_table.go:858
  | github.com/milvus-io/milvus/internal/rootcoord.(*removePartitionMetaStep).Execute
  | \t/Users/zilliz/repo/github/chyezh/milvus/internal/rootcoord/step.go:373
  | github.com/milvus-io/milvus/internal/rootcoord.(*stepStack).Execute
  | \t/Users/zilliz/repo/github/chyezh/milvus/internal/rootcoord/step_executor.go:59
  | github.com/milvus-io/milvus/internal/rootcoord.(*bgStepExecutor).process.func1
  | \t/Users/zilliz/repo/github/chyezh/milvus/internal/rootcoord/step_executor.go:201
  | runtime.goexit
  | \t/opt/homebrew/Cellar/go/1.21.1/libexec/src/runtime/asm_arm64.s:1197
Wraps: (4) collection=444883324913245356
Wraps: (5) collection not found
Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *withstack.withStack (4) *errutil.withPrefix (5) merr.milvusError"] [step="remove partition meta, collection: 444883324913245356, partition: 444883324913245361, ts: 444883843027042324"]
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by longjiquan**
fix: #27680

### CODE REVIEW

File: internal/metastore/kv/rootcoord/kv_catalog.go [hunk 1]
```diff
@@ -523,6 +523,11 @@ func dropPartition(collMeta *pb.CollectionInfo, partitionID typeutil.UniqueID) {
 
 func (kc *Catalog) DropPartition(ctx context.Context, dbID int64, collectionID typeutil.UniqueID, partitionID typeutil.UniqueID, ts typeutil.Timestamp) error {
 	collMeta, err := kc.loadCollection(ctx, dbID, collectionID, ts)
+	if err != nil && errors.Is(err, merr.ErrCollectionNotFound) {
```
Comment (chyezh) [inline internal/metastore/kv/rootcoord/kv_catalog.go:526]: `errors.Is(err, merr.ErrCollectionNotFound)` is enough.

Comment (longjiquan) [reply internal/metastore/kv/rootcoord/kv_catalog.go:526]: OK

**Review by chyezh — APPROVED** [review]

**Pull request merged**